cmd/podman/kube/generate.go

```diff
@@ -68,10 +70,32 @@ func generateFlags(cmd *cobra.Command) {
 	flags.StringVarP(&generateFile, filenameFlagName, "f", "", "Write output to the specified path")
 	_ = cmd.RegisterFlagCompletionFunc(filenameFlagName, completion.AutocompleteDefault)
 
+	typeFlagName := "type"
+	flags.StringVarP(&generateOptions.Type, typeFlagName, "t", define.PodKind, "Generate YAML for the given Kubernetes kind")
+	_ = cmd.RegisterFlagCompletionFunc(typeFlagName, completion.AutocompleteNone)
+
+	replicasFlagName := "replicas"
+	flags.Int32VarP(&generateOptions.Replicas, replicasFlagName, "r", 0, "Set the replicas number for Deployment kind")
+	_ = cmd.RegisterFlagCompletionFunc(replicasFlagName, completion.AutocompleteNone)
+
 	flags.SetNormalizeFunc(utils.AliasFlags)
 }
 
 func generateKube(cmd *cobra.Command, args []string) error {
+	typeVal, err := cmd.Flags().GetString("type")
+	if err != nil {
+		return err
+	}
+	typeVal = strings.ToLower(typeVal)
+	if typeVal != define.PodKind && typeVal != define.DeploymentKind {
+		return fmt.Errorf("invalid type given, only supported types are pod and deployment")
+	}
+	if cmd.Flags().Changed("replicas") {
+		if typeVal != define.DeploymentKind {
+			return fmt.Errorf("--replicas can only be set when type=deployment")
```

Comment:
Should we move this check to `pkg/domain/infra/abi` to make sure REST calls would also error out?  Some REST calls may not come from a `podman` binary.

Reply:
fixed